src/System.Security.Cryptography.Cng/src/System/Security/Cryptography/CngKey.Export.cs

```diff
@@ -32,10 +32,11 @@ public byte[] Export(CngKeyBlobFormat format)
                 throw errorCode.ToCryptographicException();
 
             byte[] buffer = new byte[numBytesRequired];
-            errorCode = Interop.NCrypt.NCryptExportKey(_keyHandle, IntPtr.Zero, format.Format, IntPtr.Zero, buffer, numBytesRequired, out numBytesRequired, 0);
+            errorCode = Interop.NCrypt.NCryptExportKey(_keyHandle, IntPtr.Zero, format.Format, IntPtr.Zero, buffer, buffer.Length, out numBytesRequired, 0);
```

Comment:
Ok, I think misread your question, then. Since I was calling the Length property pretty much everywhere else, I decided to make this one consistent with that too, as it’s a more robust pattern. I hate debugging native buffer overruns.